packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/task/ViewGanttDescriptionConverter.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.view.emf.task;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.components.compatibility.emf.DomainClassPredicate;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.gantt.TaskDetail;
+import org.eclipse.sirius.components.gantt.description.GanttDescription;
+import org.eclipse.sirius.components.gantt.description.TaskDescription;
+import org.eclipse.sirius.components.interpreter.AQLInterpreter;
+import org.eclipse.sirius.components.representations.GetOrCreateRandomIdProvider;
+import org.eclipse.sirius.components.representations.IRepresentationDescription;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.components.view.RepresentationDescription;
+import org.eclipse.sirius.components.view.emf.IRepresentationDescriptionConverter;
+import org.springframework.stereotype.Service;
+
+/**
+ * Converts a View-based gantt description into an equivalent {@link GanttDescription}.
+ *
+ * @author lfasani
+ */
+@Service
+public class ViewGanttDescriptionConverter implements IRepresentationDescriptionConverter {
+
+    private static final String DEFAULT_GANTT_LABEL = "Gantt";
+
+    private static final String DEFAULT_GANTT_DESCRIPTION_LABEL = "Gantt Description";
+
+    private final IObjectService objectService;
+
+    private final Function<VariableManager, String> semanticTargetIdProvider;
+
+    private final Function<VariableManager, String> semanticTargetKindProvider;
+
+    private final Function<VariableManager, String> semanticTargetLabelProvider;
+
+    private GanttIdProvider ganttIdProvider;
```

Comment:
can be final with the Objects.requireNonNull below

Reply:
done